shared/src/main/java/uk/gov/di/authentication/shared/state/conditions/TermsAndConditionsVersionNotAccepted.java

```diff
@@ -14,10 +14,18 @@ public TermsAndConditionsVersionNotAccepted(String latestVersion) {
     }
 
     @Override
-    public boolean isMet(Optional<UserProfile> context) {
+    public boolean isMet(Optional<UserContext> context) {
         if (latestVersion == null) return false;
-        return context.map(u -> !latestVersion.equals(u.getTermsAndConditions().getVersion()))
-                .orElse(true);
+        return context.map(
+                        c ->
+                                c.getUserProfile()
```

Comment:
This could probably be a 
```
context
  .flatMap(UserContent::getUserProfile)
  .map(UserProfile::getTermsAndConditions)
  .map(TermsAndConditions::getVersion)
  .filter(latestVersion::equals)
  .orElseThrow()
```

Reply:
With a minor tweak, yes! Good shout.